ledger/types/balance.go

```diff
@@ -30,211 +22,91 @@ func (b *Balance) UnmarshalBinary(data []byte) error {
 	return json.Unmarshal(data, b)
 }
 
-// IsZero returns true if all the Amount of the balance are zero.
-func (b *Balance) IsZero() bool {
-	if b == nil {
-		return true
-	}
-	return b.Balance == 0 &&
-		b.Borrowed == 0 &&
-		b.Collateral == 0 &&
-		b.Supplied == 0 &&
-		b.Virtual == 0 &&
-		b.Locked == 0
-}
-
-// Equal returns true if one Balance is equal to another.
-func (b *Balance) Equal(x *Balance) bool {
-	if b == nil || x == nil {
-		return false
-	}
-	if b.UserID != x.UserID {
-		return false
-	}
-	if b.AssetID != x.AssetID {
-		return false
-	}
-
-	return b.Balance == x.Balance &&
-		b.Borrowed == x.Borrowed &&
-		b.Collateral == x.Collateral &&
-		b.Supplied == x.Supplied &&
-		b.Virtual == x.Virtual &&
-		b.Locked == x.Locked
-}
-
-func NewBalance(userID, assetID string, balance, borrowed, collateral, supplied, virtual, locked uint64) *Balance {
-	return &Balance{
-		UserID:     userID,
-		AssetID:    assetID,
-		Balance:    balance,
-		Borrowed:   borrowed,
-		Collateral: collateral,
-		Supplied:   supplied,
-		Virtual:    virtual,
-		Locked:     locked,
+// NewBal creates a Balance.
+// If asset ID is empty or amount is negative then all fields are zero-valued and Valid() will return false.
+func NewBal(asset string, amount int64) *Balance {
+	if asset != "" && amount >= 0 {
+		return &Balance{
+			Asset:  asset,
+			Amount: uint64(amount),
+		}
 	}
+	return &Balance{}
 }
```

Comment:
Do we need this function? It just seems to be a helper to to create a balance with an int64 instead of uint64. I don't think we need this, it's easy enough to do the cast when calling `NewBalance`, and it's safer to check the amount before calling `NewBalance` rather than letting the value be lost here if it's not.